### DISCUSSION TITLE
Could not resolve module semmle.code.java.security.UnsafeDeserialization

### DISCUSSION THREAD

**Discussion by kejaly** [github/codeql#7251] [Q&A] [answered]
Excute me , i have a  problem when i use the [CWE-502/UnsafeDeserialization.ql](https://github.com/github/codeql/blob/master/java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.ql) sample code:`Could not resolve module semmle.code.java.security.UnsafeDeserialization`
![image](https://user-images.githubusercontent.com/46043605/143729255-62670c6c-c2f9-43b0-91ed-6826b3419fe6.png)
and i use the "material Icon Theme" plugin which showes a lock on the security folder：
![image](https://user-images.githubusercontent.com/46043605/143729530-75f7cd8d-9fbb-4b90-b36f-31dfeca16f39.png)

anybody can help me ?😢

**Comment by smowton** [comment]
The current version of the query imports `semmle.code.java.security.UnsafeDeserializationQuery` not `semmle.code.java.security.UnsafeDeserialization`; have you perhaps copied the `.ql` file while the `.qll` file it tries to import has been updated by Git?

**Reply by kejaly** [reply]
oh ,thanks . i solve the question by change `semmle.code.java.security.UnsafeDeserializationQuery` to `semmle.code.java.security.UnsafeDeserialization`.And i havn't copied .

**Comment by RasmusWL** [comment, marked as answer]
I noticed that they linked URL is using the `master` branch (deprecated) and not the `main` branch (the new one).

So if you use https://github.com/github/codeql/blob/main/java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.ql instead, that should solve your problem
